plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/condition/WidgetIsFound.java

```diff
@@ -96,8 +98,10 @@ public boolean test() {
 		
 		if (properWidget == null || 
 				(properWidget instanceof Control && !ControlHandler.getInstance().isVisible((Control)properWidget))) {
+			this.foundWidget = null;
 			return false;
 		}
+		this.foundWidget = properWidget;
```

Comment:
I think that here it is alright as code before returns false in case that widget was not found, right?